app/models/internal/impl/DefaultReport.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.internal.impl;
+
+import akka.actor.ActorRef;
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.metrics.portal.scheduling.Schedule;
+import com.arpnetworking.metrics.portal.scheduling.impl.NeverSchedule;
+import models.internal.reports.RecipientGroup;
+import models.internal.reports.Report;
+import models.internal.reports.ReportSource;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotNull;
+
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Default implementation of {@link Report}.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public final class DefaultReport implements Report {
+    private DefaultReport(final Builder builder) {
+        _id = builder._id;
+        _name = builder._name;
+        _schedule = builder._schedule;
+        _source = builder._source;
+        _groups = builder._groups;
+    }
+
+    @Override
+    public UUID getId() {
+        return _id;
+    }
+
+    @Override
+    public String getName() {
+        return _name;
+    }
+
+    @Override
+    public Schedule getSchedule() {
+        return _schedule;
+    }
+
+    @Override
+    public ReportSource getSource() {
+        return _source;
+    }
+
+    @Override
+    public Collection<RecipientGroup> getRecipientGroups() {
+        return Collections.unmodifiableCollection(_groups);
```

Comment:
The type should likely be one of Guava's `Immutable*` types. Then you won't need to wrap here.